app/uk/gov/hmrc/incometaxpenaltiesfrontend/controllers/IndexController.scala

```diff
@@ -38,7 +39,16 @@ class IndexController @Inject()(override val controllerComponents: MessagesContr
                                 lppCardHelper: LPPCardHelper,
                                 indexView: IndexView)(implicit appConfig: AppConfig, ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  val homePage: Action[AnyContent] = authActions.asMTDUserOldWithPenaltyData().async { implicit penaltyDataUserRequest =>
+
+
+  def homePage(isAgent:Boolean): Action[AnyContent] = authActions.asMTDUserOldWithPenaltyData().async  { implicit currentUserRequest =>
+    if(currentUserRequest.isAgent != isAgent) {
+      Future.successful(Redirect(routes.IndexController.homePage(currentUserRequest.isAgent)))
+    }else{
```

Comment:
If you call authActions.asMTDUserWithPenaltyData(isAgent) rather than the authActions.asMTDUserOldWithPenaltyData() then you won't need to do the redirect to home url stuff as the actions in there handle if an agent hits an individual endpoint and vice versa.